ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala

```diff
@@ -614,4 +632,17 @@ private[kvutils] object TransactionCommitter {
   def getContractState(commitContext: CommitContext, key: DamlStateKey): DamlContractState =
     commitContext.get(key).getOrElse(throw Err.MissingInputState(key)).getContractState
 
+  @SuppressWarnings(Array("org.wartremover.warts.Option2Iterable"))
+  private def transactionMinimumRecordTime(
+      maybeDeduplicateUntil: Option[Instant],
+      ledgerTime: Instant,
+      maxSkew: Duration,
+      submissionTime: Instant): Instant =
+    List(
+      maybeDeduplicateUntil,
+      Some(ledgerTime.minus(maxSkew)),
```

Comment:
I suggest to move the computations of the bounds derived from ledger time to the TimeModel https://github.com/digital-asset/daml/blob/master/ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/TimeModel.scala#L19 such that they are next to the `checkTime` function and you can document that these checks must be equivalent. Ideally, you'd also rewrite `checkTime` such that it uses the bounds you're computing here.

Reply:
Agreed -- it's better to keep such tightly coupled logic in the same place.

Reply:
I moved them there and also refactored `TimeModel` slightly.